### PR TITLE
remove nested structure in tests authorization

### PR DESCRIPTION
**Pull request by sofyalaski**
## Description

As commented by @sbliven in PR #1286, the extraction of token for each user is implemented within a very nested structure. This PR removes nested structure

## Motivation

Improve readability of extraction of authorization token

## Changes:
almost all test/*.js 

* test/LoginUtils.js: the first two functions `getToken` and `getIdAndToken` are rewritten to return a promise. 
* in the rest of test files the pyramid structure is replaced by async mechanism to retrieve the authorization token

## Tests included

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 

## Documentation
- [ ] swagger documentation updated \[required\]
- [ ] official documentation updated \[nice-to-have]

### CODE REVIEW

File: test/DatasetLifecycle.js [hunk 1]
```diff
@@ -12,34 +12,23 @@ var pidRaw2 = null;
 var policyIds = null;
 const raw2 = { ...TestData.RawCorrect };
 
-describe("0500: DatasetLifecycle: Test facet and filter queries", () => {
+describe.only("0500: DatasetLifecycle: Test facet and filter queries", () => {
```
Comment (Junjiequan) [inline test/DatasetLifecycle.js]: .only needs to be removed

**Review by Junjiequan — APPROVED** [review]
Nice cleanup. Just one minor issue I've found

File: test/DerivedDatasetDatablock.js [hunk 2]
```diff
@@ -10,32 +10,20 @@ describe("0750: DerivedDatasetDatablock: Test Datablocks and their relation to d
 
   let datablockId1 = null;
   let datablockId2 = null;
-
-  beforeEach((done) => {
+ 
+  beforeEach(async() => {
     before(() => {
       db.collection("Dataset").deleteMany({});
     });
```
Comment (despadam) [inline test/DerivedDatasetDatablock.js]: Shall we move this part outside of the `beforeEach()`?

File: test/DerivedDatasetOrigDatablock.js [hunk 3]
```diff
@@ -11,32 +11,20 @@ describe("0800: DerivedDatasetOrigDatablock: Test OrigDatablocks and their relat
   let origDatablockId1 = null;
   let origDatablockId2 = null;
 
-  beforeEach((done) => {
+  beforeEach(async() => {
     before(() => {
       db.collection("Dataset").deleteMany({});
       db.collection("OrigDatablock").deleteMany({});
     });
```
Comment (despadam) [inline test/DerivedDatasetOrigDatablock.js]: Also move this one?

File: test/RawDatasetDatablock.js [hunk 4]
```diff
@@ -10,31 +10,19 @@ describe("1800: RawDatasetDatablock: Test Datablocks and their relation to raw D
   var datablockId = null;
   var datablockId2 = null;
 
-  beforeEach((done) => {
+  beforeEach(async() => {
     before(() => {
       db.collection("Dataset").deleteMany({});
     });
```
Comment (despadam) [inline test/RawDatasetDatablock.js]: Move also this one?

File: test/RawDatasetOrigDatablock.js [hunk 5]
```diff
@@ -14,32 +14,20 @@ describe("2000: RawDatasetOrigDatablock: Test OrigDatablocks and their relation
     origDatablockWithEmptyChkAlg = null,
     origDatablockWithValidChkAlg = null;
 
-  beforeEach((done) => {
+  beforeEach(async() => {
     before(() => {
       db.collection("Dataset").deleteMany({});
       db.collection("OrigDatablock").deleteMany({});
     });
```
Comment (despadam) [inline test/RawDatasetOrigDatablock.js]: this one too

File: test/DatasetLifecycle.js (hunk 1 above)
Comment (sofyalaski) [reply test/DatasetLifecycle.js]: oh thanks! I always forget those..

File: test/UserAuthorization.js [hunk 6]
```diff
@@ -21,87 +21,57 @@ let accessTokenAdminIngestor = null,
...
-    );
-  });
-
+  beforeEach(async() => {
+    const loginResponseIngestor = await utils.getIdAndToken(appUrl, {
+      username: "adminIngestor",
+      password: TestData.Accounts["adminIngestor"]["password"],
+    });
+    userIdIngestor = loginResponseIngestor.userId
+    accessTokenAdminIngestor = loginResponseIngestor.token
```
Comment (despadam) [inline test/UserAuthorization.js]: Extremely minor detail but no `;` at the end of line, not only for these lines, but for all similar changes applied in this file.

File: test/Users.js [hunk 7]
```diff
@@ -38,19 +38,13 @@ describe("2350: Users: Login with functional accounts", () => {
...
-        done();
-      },
-    );
+  beforeEach(async() => {
+    const loginResponseUser1 = await utils.getIdAndToken(appUrl, {
+      username: "user1",
+      password: TestData.Accounts["user1"]["password"],
+    });
+    userIdUser1 = loginResponseUser1.userId
+    accessTokenUser1  = loginResponseUser1.token
```
Comment (despadam) [inline test/Users.js]: missing `;` at end of line

**Review by despadam — COMMENTED** [review]
Great work, I only left comments for some small details.

File: test/DerivedDatasetDatablock.js (hunk 2 above)
Comment (sofyalaski) [reply test/DerivedDatasetDatablock.js]: yes, definitely, thanks for noticing

**Review by despadam — APPROVED** [review]

**Pull request merged**